src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -705,40 +606,44 @@ public TileEntity getTileEntity()
         {
             return null;
         }
-        return this.schematicWorld.getTileEntity(this.progressPos);
+        return this.structure.getTileEntity(this.progressPos);
     }
 
     /**
-     * @return A list of all the entities in the schematic.
+     * @return A list of all the entities in the structure.
      */
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'pos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)